src/rest/zone_client.erl

```diff
@@ -60,118 +61,179 @@ index_ready(Zone) when is_binary(Zone) ->
-    {error, not_implemented}.
+is_adjacent(Zone, From, To) ->
+    {ok, Address} = http_utils:get_address(Zone),
+    Url = http_utils:build_url(Address, <<"is_adjacent">>, [{<<"from">>, From}, {<<"to">>, To}]),
+    http_executor:post_request(Url, true).
 
+% SHOULD POTENTIAL
 -spec list_adjacent(Zone:: binary(), Of:: binary()) -> {ok, list(binary())} | {error, any()}.
-list_adjacent(_Zone, _Of) ->
-    {error, not_implemented}.
+list_adjacent(Zone, Of) ->
+    {ok, Address} = http_utils:get_address(Zone),
+    Url = http_utils:build_url(Address, <<"list_adjacent">>, [{<<"of">>, Of}]),
+    http_executor:post_request(Url, true).
 
+% SHOULD POTENTIAL
 -spec list_adjacent_reversed(Zone:: binary(), Of:: binary()) -> {ok, list(binary())} | {error, any()}.
-list_adjacent_reversed(_Zone, _Of) ->
-    {error, not_implemented}.
+list_adjacent_reversed(Zone, Of) ->
+    {ok, Address} = http_utils:get_address(Zone),
+    Url = http_utils:build_url(Address, <<"list_adjacent_reversed">>, [{<<"of">>, Of}]),
+    http_executor:post_request(Url, true).
 
+% MUST POTENTIAL
 -spec permissions(Zone::binary(), From:: binary(), To:: binary()) -> {ok, binary()} | {error, any()}.
-permissions(_Zone, _From, _To) ->
-    {error, not_implemented}.
+permissions(Zone, From, To) ->
+    {ok, Address} = http_utils:get_address(Zone),
+    Url = http_utils:build_url(Address, <<"permissions">>, [{<<"from">>, From}, {<<"to">>, To}]),
+    http_executor:post_request(Url, true).
 
+% MUST
 -spec add_edge(Zone:: binary(), From:: binary(), To:: binary(), Permissions:: binary(), Trace:: binary()
     ) -> ok | {error, any()}.
-add_edge(_Zone, _From, _To, _Permissions, _Trace) ->
-    {error, not_implemented}.
-
+add_edge(Zone, From, To, Permissions, Trace) ->
+    {ok, Address} = http_utils:get_address(Zone),
+    Params = [{<<"from">>, From}, {<<"to">>, To}, {<<"permissions">>, Permissions}, {<<"successive">>, <<"false">>}]
+    ++ (case Trace of undefined -> []; _ -> [{<<"trace">>, Trace}] end),
+    Url = http_utils:build_url(Address, <<"graph/edges">>, Params),
+    http_executor:post_request(Url, false).
+
+% MUST
 -spec add_edge(Zone:: binary(), From:: binary(), To:: binary(), Permissions:: binary(), Trace:: binary(),
     Successive:: boolean()) -> ok | {error, any()}.
-add_edge(_Zone, _From, _To, _Permissions, _Trace, _Successive) ->
-    {error, not_implemented}.
+add_edge(Zone, From, To, Permissions, Trace, Successive) ->
+    {ok, Address} = http_utils:get_address(Zone),
+    Params = [{<<"from">>, From}, {<<"to">>, To}, {<<"permissions">>, Permissions}, {<<"successive">>, atom_to_binary(Successive, utf8)}]
+      ++ (case Trace of undefined -> []; _ -> [{<<"trace">>, Trace}] end),
+    Url = http_utils:build_url(Address, <<"graph/edges">>, Params),
+    http_executor:post_request(Url, false).
 
 -spec add_edges(Zone:: binary(), BulkRequest:: binary()) -> ok | {error, any()}.
-add_edges(_Zone, _BulkRequest) ->
+add_edges(Zone, BulkRequest) ->
```

Comment:
Czemu to się nie wywala przy kompilacji? Nie jest przypadkiem tak, że nieużywana funkcja musi mieć _ na początku?